tests/test_clean.py

```diff
@@ -346,3 +347,15 @@ def test_memory_usage_many_elements_with_long_tails(self):
         mem = peak_memory_usage(cleaner.clean_html, html)
 
         self.assertTrue(mem < 10, f"Used {mem} MiB memory, expected at most 10 MiB")
+
+    def test_possibly_invalid_url(self):
+        cleaner = Cleaner()
+        with warnings.catch_warnings(record=True) as w:
```

Comment:
Could you please do something like this (not tested)?

```suggestion
        with warnings.catch_warnings(record=True, cetegory=LXMLHTMLCleanWarning) as w:
```

That way, you don't have to assert the issubclass call and possible Python DeprecationWarnings will not mess with the test results but propagate normally.

Reply:
I cannot. Specifying category is not supported in Python < 3.11, see https://docs.python.org/3/library/warnings.html#available-context-managers